scripts/build.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/bash
+
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  ScriptRoot="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+  SOURCE="$(readlink "$SOURCE")"
+  [[ $SOURCE != /* ]] && SOURCE="$ScriptRoot/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+ScriptRoot="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+
+build=false
+ci=false
+configuration='Debug'
+generate=false
+help=false
+install=false
+remaining=''
+
+while [[ $# -gt 0 ]]; do
+  lower="$(echo "$1" | awk '{print tolower($0)}')"
+  case $lower in
+    --build)
+      build=true
+      shift 1
+      ;;
+    --ci)
+      ci=true
+      shift 1
+      ;;
+    --configuration)
+      configuration=$2
+      shift 2
+      ;;
+    --generate)
+      generate=true
+      shift 1
+      ;;
+    --help)
+      help=true
+      shift 1
+      ;;
+    --install)
+      install=true
+      shift 1
+      ;;
+    *)
+      if [ -z "$remaining" ]; then
+        remaining="$1"
+      else
+        remaining="$remaining $1"
+      fi
+      shift 1
+      ;;
+  esac
+done
+
+function Build {
+  if [ -z "$remaining" ]; then
+    cmake --build "$BuildDir"
+  else
+    cmake --build "$BuildDir" "${remaining[@]}"
+  fi
+
+  LASTEXITCODE=$?
+
+  if [ "$LASTEXITCODE" != 0 ]; then
+    echo "'Build' failed"
+    return "$LASTEXITCODE"
+  fi
+}
+
+function CreateDirectory {
+  if [ ! -d "$1" ]; then
+    mkdir -p "$1"
+  fi
+}
+
+function Generate {
+  if [ -z "$remaining" ]; then
```

Comment:
I was having trouble getting bash to do what I wanted here; so I just manually expanded everything out. It's not pretty, but it works for now.